### ISSUE TITLE
[expo-notifications] Invalid parsing of vibrate flag in Android FCM notifications

### ISSUE THREAD

**Issue by ghost** [expo/expo#17410]
### Summary

To show [heads-up notification](https://developer.android.com/guide/topics/ui/notifiers/notifications#Heads-up) on Android, you need to use either ringtone or vibrations. When expo is [parsing FCM notification payload](https://github.com/expo/expo/blob/main/packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/JSONNotificationContentBuilder.java#L134) it reads the `vibrate` key using following code:

```
protected boolean shouldUseDefaultVibrationPattern(JSONObject payload) {
  return !payload.optBoolean(VIBRATE_KEY, true);
}
```

This leads to following cases:

* The payload has the `vibrate: "true"`: `shouldUseDefaultVibrationPattern` returns `false`, which **does not cause vibration** (and it's not a heads-up notification)
* The payload has `vibrate: "false"`: `shouldUseDefaultVibrationPattern` returns `true`, which **does cause vibration** (and the heads-up notification)
* The payload does not have `vibrate` key: `shouldUseDefaultVibrationPattern` returns `false`, same as `vibrate: true`. 2nd arg to `optBoolean` is the default value.

The `shouldUseDefaultVibrationPattern` flag is then used to [construct native Android notification](https://github.com/expo/expo/blob/main/packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/presentation/builders/ExpoNotificationBuilder.java#L67).

Since the behaviour of the `vibrate` flag in FCM notification payload works opposite to what one could expect, I suspect the negation in `shouldUseDefaultVibrationPattern` is a bug. 

**PS.** There is a small interaction between this bug and [respect shouldPlaySound: false in setNotificationHandler fix](https://github.com/expo/expo/pull/13411/files#diff-b93bcf91d68a44b539f44f7bfe012ff30642bc02a3b7c6b44b065a7a7c9a0020L74) as some notifications have become silent (no sound) and no longer show up as heads-up notifications.  It's possible to revert this change using [patch-package](https://www.npmjs.com/package/patch-package), but that's a temporary workaround. I'm mentioning this since other devs may be looking for this fix.

**TL;DR:** `vibrate: false` in FCM notification payload **causes vibration**. `vibrate: true` FCM in notification payload **does not cause vibrations**.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### Environment

  expo-env-info 1.0.3 environment info:
    System:
      OS: macOS 11.6
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 17.2.0 - /usr/local/bin/node
      Yarn: 1.22.17 - /usr/local/bin/yarn
      npm: 6.14.15 - /usr/local/bin/npm
      Watchman: 2021.10.18.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
      Android SDK:
        API Levels: 30, 31
        Build Tools: 29.0.2, 31.0.0
        System Images: android-30 | Google Play Intel x86 Atom, android-31 | Intel x86 Atom_64, android-31 | Google APIs Intel x86 Atom_64
    IDEs:
      Android Studio: 2020.3 AI-203.7717.56.2031.7784292
      Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
    npmPackages:
      babel-preset-expo: 9.0.2 => 9.0.2
      expo: ^44.0.0 => 44.0.6
      react: 17.0.1 => 17.0.1
      react-dom: 17.0.1 => 17.0.1
      react-native: 0.64.3 => 0.64.3
      react-native-web: ^0.17.5 => 0.17.6
    npmGlobalPackages:
      eas-cli: 0.46.0
      expo-cli: 5.0.3
    Expo Workflow: bare

## Server side

```
const message = {
  to: YOUR_DEVICE_PUSH_TOKEN,
  priority: "high",
  collapse_key: "test",
  time_to_live: 30 * 60 * 1000, // 30m
  data: {
    body: {},
    title: 'MY_TEST_TITLE',
    message: 'MY_TEST_MESSAGE_BODY',
    channelId: 'test_channel_id,
    priority: "high",
    // vibrate: "true", // will NOT cause vibration
    vibrate: "false", // will cause vibration
    // 'sound' key is not required to reproduce this error
  },
};

fetch('https://fcm.googleapis.com/fcm/send', {
  method: "POST",
  body: JSON.stringify(message),
  headers: {
    'Authorization': `key=${YOUR_FCM_AUTH_KEY}`,
    'Content-Type': 'application/json'
  },
  timeout: 60 * 1000, // 1 minute
})
```

## Mobile side

App with configured `expo-notifications` using FCM on Android phone - https://docs.expo.dev/versions/latest/sdk/notifications/. We are using the most basic configuration, but the following bit is the most important:

```
Notifications.setNotificationHandler({
  handleNotification: async () => ({
    shouldShowAlert: true,
    shouldPlaySound: true, // important!
    shouldSetBadge: true,
  }),
  handleError: (err) => console.warn("handle notifications error", err)
});
```

**Comment by michalv8** [comment]
I am having similar issue. But I'm still using Expo SDK in version 44 - as version 45 have been released a couple of weeks ago -  @MarMatLC have you, maybe updated your project to 45 and reproduced this issue (or has it been fixed)?
